### ISSUE TITLE
Losing cache after HOOBS restart

### ISSUE THREAD

**Issue by Stetocasa** [bwp91/homebridge-ewelink#219]
Hi Ben, plug-in and hoobs updated to last version.
Every single hoobs restart, each EWeLink accessories come back to simple switch losing name, type and automation.

**Comment by bwp91** [comment]
Looks like this *might* be an issue with the hoobs update...

https://github.com/hoobs-org/HOOBS/issues/1149

https://github.com/hoobs-org/HOOBS/issues/1150

**Comment by bwp91** [comment]
does this happen **every** time you restart hoobs? or just the once after initially updating hoobs / this plugin?

**Comment by apartmedia** [comment]
It seems to happen at every reboot. Each reboot all rooms are getting lost and all accessories are dumped into the HOOBS Bridge room in Home App.

**Comment by apartmedia** [comment]
How much space is needed for the update? I have a 8GB SD card and after upgrading to 3.3.4 suddenly 97% is used, just 260MB left free. May be the update kept a lot of old files and could not fully complete or there is not enough space left for writing everything needed into the database?
Only guessing...

**Comment by Stetocasa** [comment]
Reboot multiples plug-in, but a simplest restart the EWeLink ones also lost

**Comment by bwp91** [comment]
I believe this is an issue with the latest HOOBS update. Please wait for them to fix the issue with a further update.

**Comment by Stetocasa** [comment]
I'm sure reading a lot of comments on that. Please close this plug-in issue.

**Comment by bwp91** [comment]
You can downgrade to hoobs 3.3.3 by STOPPING the hoobs service then running these commands separately via SSH:

```
sudo npm uninstall -g @hoobs/hoobs
sudo npm install -g --unsafe-perm @hoobs/hoobs@3.3.3
```

**Comment by apartmedia** [comment]
> You can downgrade to hoobs 3.3.3 by STOPPING the hoobs service then running these commands separately via SSH:
> 
> ```
> sudo npm uninstall -g @hoobs/hoobs
> sudo npm install -g --unsafe-perm @hoobs/hoobs@3.3.3
> ```

Do you know if this kind of a downgrade will rescue the database and room assignments or will reassigning everything by hand again being definitely necessary at any case?

**Comment by Stetocasa** [comment]
I will wait next days to see if hoobs will release an update fix
In case not early, I will follow your downgrade suggestion

**Comment by bwp91** [comment]
> Do you know if this kind of a downgrade will rescue the database and room assignments or will reassigning everything by hand again being definitely necessary at any case?

The downgrade won't recover previous homekit names, room assignments, etc. By hand will almost certainly be the case.

**Comment by bwp91** [comment]
can anyone confirm that hoobs 3.3.5 fixes this? then i will get this issue closed.
thanks!

**Comment by apartmedia** [comment]
> can anyone confirm that hoobs 3.3.5 fixes this? then i will get this issue closed.
> thanks!

In general yes, for me just updating did not the trick. As 3.3.4 corrupted the accessory cache I had to remove or renaming them via SSH. After a reboot I could rearrange everything. Further reboots did not messing up the rooms and accessories anymore. So seems to be fixed.
But after all this I wasn‘t able to update some plugins anymore. Tried this with updating Alexa 0.5.30 to 0.5.32 which caused stopping the server, surprisingly with an error message about a totally different plugin (first in the config, Shelly in my case).
I had to delete all my plugins and reinstalling them one by one. After that everything seemed working also with latest plugin versions.
